### ISSUE TITLE
Remote gdb debugging doesn't work

### ISSUE THREAD

**Issue by afedorychak** [radare/radare2#4507]
When I connect to a gdbserver via `gdb` I can see memory maps, it also automatically loads all the symbols from my `*.so` files and somehow maps them to actual memory locations which is pretty handy.

How can I achieve the same result with radare? After attaching it shows registers correctly, but memory and backtrace is not available. Then after a few innocuous commands it dies completely.

```
$ radare2 -a x86 -b 32 -D gdb gdb://localhost:1337 
attach 7 7
attach -999956960 1
r_debug_gdb_reg_read: small buffer 176 vs 312
r_debug_gdb_reg_read: small buffer 176 vs 312
 -- Everytime you run radare2, a random file is removed :)
r_debug_gdb_reg_read: small buffer 176 vs 312
[0xb7525435]> dr
r_debug_gdb_reg_read: small buffer 176 vs 312
eax = 0xfffffffc
ecx = 0xbfb76560
edx = 0x00000010
ebx = 0x0000000f
esp = 0xbfb76474
ebp = 0x00000008
esi = 0x000003e8
edi = 0x00000000
eip = 0xb7525435
eflags = 0x00200246
fctrl = 0x0000037f
fstat = 0x00004100
ftag = 0x0000ffff
fiseg = 0x00000000
fioff = 0xb6fd8847
foseg = 0x00000000
fooff = 0xbfb75a54
fop = 0x00000000
r_debug_gdb_reg_read: small buffer 176 vs 312
[0xb7525435]> dm
r_debug_gdb_reg_read: small buffer 176 vs 312
[0xb7525435]> bt
r_debug_gdb_reg_read: small buffer 176 vs 312
[0xb7525435]> fs
0   18 . regs
r_debug_gdb_reg_read: small buffer 176 vs 312
[0xb7525435]> pdf
Cannot find function at 0xb7525435
r_debug_gdb_reg_read: small buffer 176 vs 312
[0xb7525435]> 
```

After a few more `ds` and `dc` 

```
[0xb74c42d2]> dc
unpack: Packet too long
read_packet: unpack failed
r_debug_gdb_reg_read: small buffer 176 vs 624
*** Error in `radare2': free(): corrupted unsorted chunks: 0x00007f6ec465f680 ***
*** Error in `radare2': malloc(): memory corruption: 0x00007f6ec465fea0 ***
```

My version:
```
$ radare2 -version
radare2 0.10.2-git 10777 @ linux-little-x86-64 git.0.10.1-322-gfa4edbf
commit: fa4edbfa452cc9d7514cfeffa91943872c363890 build: 2016-03-31
```

**Comment by radare** [comment]
works for me on osx. can you try with this line?
```
 r2 -a x86 -b 32 -d gdb://localhost:1337 
```

**Comment by afedorychak** [comment]
```
$ r2 -a x86 -b 32 -d gdb://localhost:1337
attach 7 7
attach -385220512 -385220512
r_debug_reg: error reading registers
r_debug_reg: error reading registers
 -- Don't waste your time
r_debug_reg: error reading registers
[0x00000000]>
```

On the target:
```
# gdbserver --version                                       
GNU gdbserver (GDB) 7.6
Copyright (C) 2013 Free Software Foundation, Inc.
gdbserver is free software, covered by the GNU General Public License.
This gdbserver was configured as "i686-linux-android"

# gdbserver --debug --remote-debug :1337 --attach 8950
... see gdbserver.log ...
```
[logs.zip](https://github.com/radare/radare2/files/211540/logs.zip)

Also included the traffic captured on port `1337` (just in case).

**Comment by radare** [comment]
Works for me, but will investigate more after release

**Comment by radare** [comment]
ping. is this still happening to you?

**Comment by radare** [comment]
ping

**Comment by jduck** [comment]
I can confirm that the gdbserver support doesn't seem to work properly. I attached it to a program that just crashes in main (strcpy(buf, argv[1]) with null argv[1]) and it tell me anything. That said, "pd 10" did show me the crashing instruction and inspecting the registers did seem to work. It's concerning that it would call the PID "7" or some huge negative number.

**Comment by radare** [comment]
Well. it is working but it could be better (more verbose), closing for now

**Comment by jduck** [comment]
Perhaps I should create more specific tickets to parts of the functionality that was not working.

For example I did "sr SP; V" and scrolled up and down. For some reason the contents of the stack changed as I scrolled... No way... And then the top 256 bytes or so were always zero??? This did not happen with the native debugger.

There were other commands that I tried that seemed to do nothing. I suspect anyone who tries to use this thing will either run and hide or end up opening tickets like this one hehe.
